packages/blockchain-utils-linking/src/ethereum.ts

```diff
@@ -61,25 +61,18 @@ export class EthereumAuthProvider implements AuthProvider {
     return createLink(did, accountId, this.provider, this.opts)
   }
 
-  async requestCapability(params: OcapParams): Promise<Ocap> {
+  async requestCapability(siweMessage: SiweMessage): Promise<Cacao> {
```

Comment:
```suggestion
  async requestCapability(sessionDID: string, streams: Array<StreamId | string>): Promise<Cacao> {
```
I would like to simplify the method a lot here. Really most things inside of the SIWE message should be provided by this method!

Reply:
uhh i dont agree.

Nonces should be managed by the dApp, they could use them for linking to other information.
Expiry and not before times should be managed by the dApp
the statement and domain should be managed by the dApp

any defaults we choose here will be opinionated for our use case for 3IDConnectV2 but for devs looking to implement their own capabilities, especially down the road with delegation, it will become a problem

Reply:
If apps are looking to link additional information / their own capabilities then there is no need for them to use this library. As it stands right now the only thing this method really does is sign the message and output a CACAO. That's doesn't provide much meaningful help to a dev. 

The purpose of this library is really an easy entrypoint for devs. However, we could provide an optional param like so:
```suggestion
  interface CapOpts {} // optional parameters to change our defaults for SIWE
  async requestCapability(sessionDID: string, streams: Array<StreamId | string>, opts: CapOpts): Promise<Cacao> {
```

The defaults I think we should provide:
```js
new SiweMessage({
      domain: window ? window.location.hostname : throw Error(...)
      address: address, // from provider obv
      statement: 'Give the current application access to some of your data on Ceramic', // copy can be updated later
      uri: sessionDID,
      version: '1',
      nonce: random(),
      issuedAt: // now
      expires: // one week, or some other sensible default
      chainId: // from provider
      resources: streams
    })
```

Reply:
If we want to avoid confusion, we can wrap SiweMessage in a different container which better reflects the use case. How "Sign in with Ethereum" relates to capabilities is a little involved. Why 'uri' is a DID is also a little involved. I like the suggestion of `sessionDID`, but I don't want to enforce defaults that may not work for devs.

Reply:
I like the options approach much better. Will do.

Reply:
Ok, great :)